hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1196,7 +1196,11 @@ public void abortOutstandingMultipartUploads(long seconds)
    */
   @Override
   public String getScheme() {
-    return "s3a";
+    if (this.uri != null && this.uri.getScheme() != null){
```

Comment:
just add a field "scheme" with default value FS_S3A; in initialize() set it to the URI scheme